### ISSUE TITLE
[BUG] Azure.AI.DocumentIntelligence1.0.0-beta.2 (preview), doesn't support specifying ServiceVersion thus no support for older REST API versions

### ISSUE THREAD

**Issue by jaliyaudagedara** [Azure/azure-sdk-for-net#43415]
### Library name and version

Azure.AI.DocumentIntelligence 1.0.0-beta.2 (preview)

### Describe the bug

[DocumentIntelligenceClientOptions](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/documentintelligence/Azure.AI.DocumentIntelligence/src/Generated/DocumentIntelligenceClientOptions.cs) doesn't support older `ServiceVersions`.

![image](https://github.com/Azure/azure-sdk-for-net/assets/5653381/a497b76e-68a9-45c3-8ff9-657da4ff3490)

However, the [documentation](https://learn.microsoft.com/en-us/azure/ai-services/document-intelligence/sdk-overview-v4-0?view=doc-intel-4.0.0&tabs=csharp) states it is supported.

![image](https://github.com/Azure/azure-sdk-for-net/assets/5653381/d9e1006d-2124-4806-96a1-046ebb9c9a02)

### Expected behavior

When constructing `DocumentIntelligenceClient` or `DocumentIntelligenceAdministrationClient`,  we can pass in `DocumentIntelligenceClientOptions`, however right now we can't specify the `ServiceVersion`. We should be able to specify the `ServiceVersion`.

### Actual behavior

`Azure.AI.FormRecognizer` supports specifying the `ServiceVersion` version when constructing a client. We don't have the same behavior in `Azure.AI.DocumentIntelligence`.

### Reproduction Steps

N/A

**Comment by kinelski** [comment]
Hello @jaliyaudagedara.

What you describe is the intended behavior. Going forward, `Azure.AI.DocumentIntelligence` will only support new REST API versions while our older package, `Azure.AI.FormRecognizer`, should be used to target older versions.

Note that `Azure.AI.FormRecognizer` won't get support for new REST API versions, so you'll need to migrate to `Azure.AI.DocumentIntelligence` if you ever decide to target new versions of the service. You can check our [Migration Guide](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/documentintelligence/Azure.AI.DocumentIntelligence/MigrationGuide.md) for more details.

**Comment by jaliyaudagedara** [comment]
@kinelski,

Thanks for the reply. 

I am using `Azure.AI.DocumentIntelligence`. I am sorry, I am a bit confused. 

### The expected behavior is,
When constructing `DocumentIntelligenceClient` or `DocumentIntelligenceAdministrationClient`, we can pass in `DocumentIntelligenceClientOptions` and we should be able to specify the `ServiceVersion`.

### Actual behavior as of now,
We can't specify `ServiceVersion` when using `DocumentIntelligenceClientOptions` in `Azure.AI.DocumentIntelligence`

**Comment by kinelski** [comment]
@jaliyaudagedara,

Thanks for bringing this to our attention. Unfortunately, the piece of documentation you shared that lists supported API versions is incorrect. I'll get in touch with our documentation team and ask them to get it fixed.

Originally the Document Intelligence service was named Form Recognizer and, similarly, we had an [Azure.AI.FormRecognizer](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/formrecognizer/Azure.AI.FormRecognizer/README.md) SDK package, which supports every GA service version released up to today.

After the service was rebranded to Document Intelligence, we shifted development to a new [Azure.AI.DocumentIntelligence](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/documentintelligence/Azure.AI.DocumentIntelligence/README.md) package. It's still in beta stage and the only REST API version it supports today is the latest preview, 2024-02-29-preview, but it will offer support for any version released from now on.

To summarize, you should use the old `Azure.AI.FormRecognizer` package if you intend to target REST API version v2.0, v2.1, 2022-08-31, or 2023-07-31; or use the new `Azure.AI.DocumentIntelligence` package if you intend to target any REST API version newer than 2023-07-31.

If there's no particular need to target older service versions, we strongly recommend sticking to the new Document Intelligence package as it will be the one to receive new features and updates going forward.

**Comment by jaliyaudagedara** [comment]
@kinelski 

Thanks for the reply. And yes please, we need to get the documentation updated, it was the one that was throwing me off.

And we are always on the latest versions, we don't use `Azure.AI.FormRecognizer`, that's dead to us :)

Thanks again!

**Comment by jaliyaudagedara** [comment]
Re-opened as this can be used to link the documentation update PR.

**Comment by jaliyaudagedara** [comment]
@kinelski, I have updated the documentation and linked the issue.

**Comment by kinelski** [comment]
@jaliyaudagedara Thank you for taking the initiative! We can close this issue once the documentation is updated.

**Comment by kinelski** [comment]
Closing this issue now that the documentation has been updated.

**Comment by jaliyaudagedara** [comment]
Thanks @kinelski!
